socorro/unittest/lib/test_javautil.py

```diff
@@ -0,0 +1,128 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from socorro.lib import javautil
+
+
+EXC = """\
+Exception: msg
+\tat org.File.function(File.java:100)
+\tat org.File.function2(File.java:200)\
+"""
+
+
+def test_parse_basic():
+    """Parse a basic exception with a class, message, and some stack lines"""
+    java_exc = javautil.parse_java_stack_trace(EXC)
+    assert java_exc.exception_class == 'Exception'
+    assert java_exc.exception_message == 'msg'
+    assert java_exc.stack == [
+        'at org.File.function(File.java:100)',
+        'at org.File.function2(File.java:200)'
+    ]
+    assert java_exc.additional == []
+
+
+EXC_WITH_MULTILINE_MSG = """\
+android.database.sqlite.SQLiteDatabaseLockedException: database is locked (code 5)
+#################################################################
+Error Code : 5 (SQLITE_BUSY)
+Caused By : The database file is locked.
+\t(database is locked (code 5))
+#################################################################
+\tat android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
+\tat android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:904)
+"""
```

Comment:
This exception is fantastic in that it has a tabbed line that's not part of the stack and it has a multi-line message. Yay!